### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by Braingenius2**
# React & Redux group project - Space Travelers' Hub

## Summary

This pull request contains the final submission of the Space Travelers' Hub project, which is a web application built using React and Redux. The project allows users to book rockets, join space missions, and view their profile with reserved rockets and joined missions. The project fulfils all the specified learning objectives, requirements, and optional tasks. The team worked collaboratively on the project, and each member contributed to various tasks as outlined in the Kanban board.

## Changes made

We have implemented the following features and fulfilled all requirements during the development of the Space Travelers' Hub project:

-     Created React App with React Redux and React Router.
-     Fetched data from SpaceX API and updated the Redux store with rockets, missions, and dragons (if applicable).
-     Rendered lists of rockets, missions, and dragons (if applicable) using React components and conditional rendering based on reserved/joined status.
-     Implemented actions and reducers to handle booking/cancelling rockets/dragons and joining/leaving missions.
-     Styled the application according to the provided designs using either manual styling or React Bootstrap library.
-     Created the My Profile section to display reserved rockets and joined missions.
-     Added unit tests using React Testing Library and Jest for components and functions.

## Code Review

We conducted code reviews throughout the development process, with each task having a separate pull request reviewed by a teammate. The code review process helped us maintain code quality and identify potential improvements.

## Conclusion

The Space Travelers' Hub project has been completed successfully, meeting all the required objectives and optional tasks. We have thoroughly tested the application, and it is ready for submission. Thank you for the opportunity to work on this project, and we look forward to receiving feedback during the code review process.

Contributors: @Braingenius2 , @Luffytaro22 and @PedroH-Neves

### CODE REVIEW

**Review by AbayomiOlaoye — APPROVED** [review]
Check the comment below...

File: src/App.js [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
-      </Routes>
+      <div>
+        <NavBar />
+        <Routes>
+          <Route index path="/" element={<Rockets />} />
+          <Route path="/missions" element={<Missions />} />
+          <Route path="/profile" element={<MyProfile />} />
+          <Route path="/dragons" element={<Dragons />} />
+        </Routes>
+      </div>
```
Comment (AbayomiOlaoye) [inline src/App.js:20]: - [ ] Kindly include unit tests to all your logic in order to fully accomplish the goal of the project.
Screenshot from 2023-07-29 15-56-34

**Review by AbayomiOlaoye — CHANGES_REQUESTED** [review]
## STATUS: CHANGE REQUIRED ♻️ 

Hi @Team :wave: :wave: , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

so_close

## Highlights
- Descriptive Pull Request message :heavy_check_mark: 
- Good use of GitHub flow :heavy_check_mark: 
- Clean and functional code :heavy_check_mark: 
- Meaningful commit messages :heavy_check_mark: 
- No linter error :heavy_check_mark: 
- Descriptive ReadMe documentation :heavy_check_mark: 


## Suggested changes
- [ ] _Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by gemmen29 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**